rust/src/client.rs

```diff
@@ -0,0 +1,74 @@
+use std::os::unix::net::UnixStream;
+use std::io::{self, Write, Read};
+use std::str;
+
+pub struct Client {
+    stream: UnixStream,
+}
+
+impl Client {
+    pub fn connect(socket_addr: &str) -> std::io::Result<Self> {
+        let stream = UnixStream::connect(socket_addr)?;
+        Ok(Self { stream })
+    }
+
+    pub fn set(
+        &mut self,
```

Comment:
guessing need to be mut cuz stream write needs it?

Reply:
Yep